applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp

```diff
@@ -706,14 +709,15 @@ namespace Kratos
             // We set the database
             auto& r_process_info = rModelPart.GetProcessInfo();
             ModelPart& r_computing_contact_model_part = rModelPart.GetSubModelPart("ComputingContact");
-            for (auto& r_slave_cond : slave_conds) {
-                for (auto& r_master_cond : master_conds) {
-                    r_cond++;
-                    Condition::Pointer p_auxiliar_condition = r_computing_contact_model_part.CreateNewCondition("ALMFrictionalMortarContactCondition3D4N", r_cond, r_slave_cond->GetGeometry(), p_cond_prop_0);
+            for (auto p_slave_cond : slave_conds) {
+                for (auto p_master_cond : master_conds) {
+                    id_cond++;
+                    const PairedCondition& r_reference_condition = dynamic_cast<const PairedCondition&>(KratosComponents<Condition>::Get("ALMFrictionalMortarContactCondition3D4N"));
+                    Condition::Pointer p_auxiliar_condition = r_reference_condition.Create(id_cond, p_slave_cond->pGetGeometry(), p_cond_prop_0, p_master_cond->pGetGeometry());
                     // We set the geometrical values
-                    p_auxiliar_condition->SetValue(PAIRED_GEOMETRY, r_master_cond->pGetGeometry());
-                    p_auxiliar_condition->SetValue(NORMAL, r_slave_cond->GetValue(NORMAL));
-                    p_auxiliar_condition->SetValue(PAIRED_NORMAL, r_master_cond->GetValue(NORMAL));
+                    r_computing_contact_model_part.AddCondition(p_auxiliar_condition);
+                    p_auxiliar_condition->SetValue(NORMAL, p_slave_cond->GetValue(NORMAL));
```

Comment:
Can't you get this from the geometry?

Reply:
The normal?, I could do it, but there are corner cases...

Reply:
Ah, alright. Maybe I have something for this, later...